sdk/core/core-lro/src/poller.ts

```diff
@@ -0,0 +1,133 @@
+import { EventEmitter } from "events";
+import { delay, HttpOperationResponse, RequestOptionsBase, ServiceClient } from "@azure/core-http";
+
+export interface PollerOptionalParameters<States> {
+  manual?: boolean;
+  intervalInMs?: number;
+  state?: States;
+  client?: ServiceClient;
+  initialResponse?: HttpOperationResponse;
+  previousResponse?: HttpOperationResponse;
+  requestOptions?: RequestOptionsBase;
+  stopped?: boolean;
+}
+
+export abstract class Poller<States, Result> extends EventEmitter  {
```

Comment:
I'm torn on if EventEmitter is too heavy for the events we have. Certainly `done` and `pollError` could be supplanted with the promise returned by `done()`, so really `newResponse` is the only interesting thing.

Reply:
Wouldn't a state update be interesting too?

Reply:
I'm not sure newResponse is interesting except in the context of after the state gets updated anyway, so "update" is one generic event for me.